src/db/mysql/models.rs

```diff
@@ -524,12 +532,12 @@ impl MysqlDb {
             Sorting::Oldest => query.order(bso::id.asc()).order(bso::modified.asc()),
             */
             Sorting::Index => query.order(bso::sortindex.desc()),
-            Sorting::Newest => query.order(bso::modified.desc()),
-            Sorting::Oldest => query.order(bso::modified.asc()),
+            Sorting::Newest => query.order((bso::modified.desc(), bso::id.desc())),
```

Comment:
Yes. It's all so clear to me now. So to do the LIMIT/OFFSET removal optimization, we'll need to pass the timestamp:id pair instead of timestamp:offset pair, and skip over any records matching that timestamp that don't have the same id.